src/walk.ts

```diff
@@ -25,18 +25,34 @@ export type Options = {
  * @returns An array of absolute paths of all parent directories.
  */
 export function up(base: string, options?: Options): string[] {
-	let { stop, cwd } = options || {};
+	if (typeof options?.stop === 'undefined') {
+		let tmp = absolute(base, options?.cwd);
+		let prev: string;
+		const arr: string[] = [];
 
-	let tmp = absolute(base, cwd), root = !stop;
-	let prev: string, arr: string[] = [];
+		do {
```

Comment:
do-while vs while, strictly preference.

Reply:
Not really
It skips the first time it does the condition and runs the code immediately

Reply:
It passes all tests

Reply:
So I assume my condition is correct and that skip is correct